### PR TITLE
ci/dependabot updates

### PR DESCRIPTION
**Pull request by cpanato**
- make sure CI install the correct go
- group dependabot updates

/assign @aojea @BenTheElder

### CODE REVIEW

**Comment by BenTheElder** [comment]
/hold

we can't just upgrade to 1.22 because of the toolchain changes

there's a pending PR in #3335 that resolves those

File: .go-version [hunk 1]
```diff
@@ -1 +1 @@
-1.20.13
+1.22.1
```
Comment (BenTheElder) [inline .go-version:1]: this will actually cause us to use go 1.16 because of toolchains :-)

**Comment by BenTheElder** [comment]
can you rebase this to remove the merge commit? it will make bisecting less confusing.

we could do squash merging, except this PR has multiple unrelated changes

**Comment by cpanato** [comment]
> can you rebase this to remove the merge commit? it will make bisecting less confusing.
> 
> we could do squash merging, except this PR has multiple unrelated changes

that was weird, i swear i did a rebase this morning, but i can be wrong, missing coffee lol

but rebase is done now :)

**Comment by BenTheElder** [comment]
thanks.

Seems we shouldn't switch to 1.22 yet: https://github.com/opencontainers/runc/pull/4240/files https://kubernetes.slack.com/archives/CHGFYJVAN/p1712086885304549

**Comment by cpanato** [comment]
we can hold or close this pr

**Comment by BenTheElder** [comment]
sigh: https://github.com/kubernetes-sigs/kind/pull/3335#issuecomment-2035083833

maybe we can merge the rest without the go bump?

we may have to roll back go temporarily even for 1.21

**Comment by cpanato** [comment]
@BenTheElder PTAL

**Comment by BenTheElder** [comment]
cc @aojea 
sorry, I'm out of time / closing tabs on my way to vacation, got distracted by broken build, security patching, etc...

**Comment by aojea** [comment]
we already setup go from the makefile

https://github.com/kubernetes-sigs/kind/blob/915b13a2d3571b26b50fd486c6c500311f1dec3f/Makefile#L37

https://github.com/kubernetes-sigs/kind/blob/915b13a2d3571b26b50fd486c6c500311f1dec3f/hack/build/setup-go.sh#L19-L40

I assume this does not harm and saves some time

/lgtm
/approve

**Pull request merged**